### PR TITLE
Add --deny flag support

### LINKED ISSUE #1580: --deny: need CLI option to auto-deny permission prompts
probably should be `--deny`

### PR DESCRIPTION
**Pull request by sh7dm**
Fixes #1580.
Testing:
`python tools/test.py`
To test:
This should throw an error: `deno https://deno.land/x/http/file_server.ts --deny`
This should work: `deno https://deno.land/x/http/file_server.ts --allow-net --deny`
This should also work: `deno https://deno.land/x/http/file_server.ts -A --deny`

### CODE REVIEW

**Comment by bartlomieju** [comment]
> This should work: `deno https://deno.land/x/http/file_server.ts --allow-net --deny`
> This should also work: `deno https://deno.land/x/http/file_server.ts -A --deny`
It defeats purpose of explicit permissions. Why would I enable access to net and deny all and still get net permission?

**Comment by sh7dm** [comment]
@bartlomieju Fixed

**Comment by bartlomieju** [comment]
I actually meant that I see no purpose for this flag, but let's wait for opinion of others.

**Comment by sh7dm** [comment]
@ry what did you mean in #1580? Which one is correct: 9360400 or 2e99561?

File: src/permissions.rs [hunk 1]
```diff
@@ -24,13 +25,17 @@ impl DenoPermissions {
...
+      deny: flags.deny
     }
   }
 
   pub fn check_run(&self) -> DenoResult<()> {
     if self.allow_run.load(Ordering::SeqCst) {
       return Ok(());
     };
+    if self.deny {
+      return Err(permission_denied());
```
Comment (hayd) [inline src/permissions.rs]: perhaps this error should be inside permission_prompt?

Comment (hayd) [reply src/permissions.rs]: i.e. move permission_prompt to a method and add `|| self.deny` to line 101.

**Comment by hayd** [comment]
Use the existing permission_prompt tests https://github.com/denoland/deno/blob/master/tools/permission_prompt_test.py

File: src/permissions.rs [hunk 2]
```diff
@@ -15,6 +15,7 @@ pub struct DenoPermissions {
   pub allow_net: AtomicBool,
   pub allow_env: AtomicBool,
   pub allow_run: AtomicBool,
+  deny: bool,
```
Comment (ry) [inline src/permissions.rs]: Make this pub too

Comment (sh7dm) [reply src/permissions.rs]: @ry done!

**Comment by hayd** [comment]
Should we test e.g. that --allow-net --deny on a net test passes?

**Comment by sh7dm** [comment]
@hayd I'll do it!

**Comment by sh7dm** [comment]
@hayd done

**Comment by ry** [comment]
This is an old PR. We have --no-prompt which handles this situation now.

Thanks @sh7dm. Closing.

**Pull request closed without merging**